### PR TITLE
Fabric

### PR DESCRIPTION
**Pull request by leahnp**
Migrated fabric stanza for #277

### CODE REVIEW

File: ansible/roles/kraken.fabric/kraken.fabric.flannel/tasks/deploy-canal.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  register: kube_networking_namespace
+
+- name: Confirm kube-networking namespace created
+  fail:
+    msg: "kube-networking namespace creation failed"
+  when: not kube_networking_namespace.rc == 0
+
+- name: Deploy canal configuration
+  command: >
+    kubectl --kubeconfig={{ kubeconfig | expanduser }} apply -f {{ config_base | expanduser }}/{{ kraken_config.cluster }}/fabric/config.yaml
```
Comment (davidewatson) [inline ansible/roles/kraken.fabric/kraken.fabric.flannel/tasks/deploy-canal.yaml]: s/kraken_config.cluster/cluster.name/g

Comment (leahnp) [inline ansible/roles/kraken.fabric/kraken.fabric.flannel/tasks/deploy-canal.yaml]: that `here` link isn't showing me anything - are you referring to: ansible/roles/kraken.fabric/kraken.fabric.selector/meta/main.yaml ?

File: ansible/roles/kraken.fabric/kraken.fabric.selector/meta/main.yaml [hunk 2]
```diff
@@ -6,12 +6,12 @@ galaxy_info:
   min_ansible_version: 1.2
 
 dependencies:
-  - { role: '{{ playbook_dir }}/roles/kraken.fabric/kraken.fabric.flannel', when: kraken_config.fabric.provider == 'flannel' }
-  - { role: '{{ playbook_dir }}/roles/kraken.fabric/kraken.fabric.calico', when: kraken_config.fabric.provider == 'calico' }
+  - { role: '{{ playbook_dir }}/roles/kraken.fabric/kraken.fabric.flannel', when: cluster.fabricConfig.provider == 'flannel' }
```
Comment (davidewatson) [inline ansible/roles/kraken.fabric/kraken.fabric.selector/meta/main.yaml]: This code is in the role and the cluster fact has not been set. @coffeepac's WIP [PR](https://github.com/samsung-cnct/k2/pull/332/files#diff-34f32660eae208198409627cf229b780) shows a good way to do this.

**Comment by leahnp** [comment]
@davidewatson can you take a look?

File: ansible/roles/kraken.fabric/kraken.fabric.selector/tasks/main.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+---
+- name: Canal Fabric Type
+  include_role:
+    name: '{{ playbook_dir }}/roles/kraken.fabric/kraken.fabric.flannel'
+  with_subelements:
+     - "{{ kraken_config.clusters }}"
+     - nodePools
```
Comment (coffeepac) [inline ansible/roles/kraken.fabric/kraken.fabric.selector/tasks/main.yaml]: we don't need to include nodePools here as they don't get used in this role.  you can use the same method that you have in main.yaml.

one piece of caution though, when I was moving the provider stanza I was getting a warning about using 'clusters' as a variable in the selector task so I used a different variable name then renamed back to cluster: https://github.com/coffeepac/k2/blob/425f59b824e3ae0e2c6ea2e7b80f9847770a7354/ansible/roles/kraken.provider/kraken.provider.selector/tasks/main.yaml

**Comment by coffeepac** [comment]
one nit, otherwise LGTM

**Comment by leahnp** [comment]
@davidewatson @coffeepac I think this is good now -

**Comment by coffeepac** [comment]
LGTM, once @davidewatson approves G2G

**Pull request merged**